kernel/sched.c

```diff
@@ -767,14 +767,44 @@ bool z_thread_prio_set(struct k_thread *thread, int prio)
 		need_sched = z_is_thread_ready(thread);
 
 		if (need_sched) {
+#if defined(CONFIG_SMP)
+			int  old_prio;
+
+			old_prio = thread->base.prio;
+#endif /* CONFIG_SMP */
 			/* Don't requeue on SMP if it's the running thread */
```

Comment:
This comment is left dangling.  This new logic is doing a LOT more than just checking for _current.

Reply:
Oops. Will fix.